internal/crypto/password.go

```diff
@@ -112,19 +107,23 @@ func ParseFirebaseScryptHash(hash string) (*FirebaseScryptHashInput, error) {
 		return nil, fmt.Errorf("crypto: Firebase scrypt hash has invalid n parameter %q %w", n, err)
 	}
 	if memoryPower == 0 {
-		return nil, fmt.Errorf("crypto: Firebase scrypt hash has invalid n parameter %q: must be greater than 0", n)
+		return nil, fmt.Errorf("crypto: Firebase scrypt hash has invalid n=0")
 	}
-	// Exponent is passed in
-	memory := uint64(1) << memoryPower
 	rounds, err := strconv.ParseUint(r, 10, 64)
 	if err != nil {
 		return nil, fmt.Errorf("crypto: Firebase scrypt hash has invalid r parameter %q: %w", r, err)
 	}
+	if rounds == 0 {
```

Comment:
All for the additional safeguards, and defensive programming given this is a critical section for the auth system.

However, in  case we ever end up digging into this section, here's were some of the initial considerations when going through the code

the current version of the library should panic if `rounds == 0` or `threads=0`

https://cs.opensource.google/go/x/crypto/+/refs/tags/v0.31.0:scrypt/scrypt.go;l=133

on this line since 
`in[(2*r-1)*16:]` would equate to `in[-16]`


For `threads=0` it should also return a division by 0 error :
```
if uint64(r)*uint64(p) >= 1<<30 || r > maxInt/128/p || r > maxInt/256 || N > maxInt/128/r {
		return nil, errors.New("scrypt: parameters are too large")
	}
```